src/tracking/envelope.cpp

```diff
@@ -137,7 +137,8 @@ namespace impactx
                 }, element_variant);
 
                 // sub-steps for space charge within the element
-                for (int slice_step = 0; slice_step < nslice; ++slice_step)
+                int nsteps = std::round(nslice/2.0);
+                for (int slice_step = 0; slice_step < nsteps; ++slice_step)
```

Comment:
I think we could even do this: replace the for loop with an iterator ("generator") of slices.

The generator we can implement similar to different integrators in separate files and that way we can provide multiple strang splitting options.

The code here would then read something like:
```c++
auto s_slicer = StrangSplitter(order, nsteps);
for (auto & slice : s_slicer)
```

Reply:
Discussed today: does not work so easily, because  for higher-order schemes we also need to split the weights and number of calls to pushes inside the slice loop body. Will revert the commit that added the strang splitter.